qiskit_nature/problems/second_quantization/lattice/models/ising_model.py

```diff
@@ -0,0 +1,117 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Ising model"""
+from typing import cast, Optional
+
+import numpy as np
+
+from qiskit_nature.operators.second_quantization import SpinOp
+from qiskit_nature.problems.second_quantization.lattice.lattices import Lattice
+
+from .lattice_model import LatticeModel
+
+
+class IsingModel(LatticeModel):
+    """The Ising model."""
+
+    @classmethod
+    def uniform_parameters(
+        cls,
+        lattice: Lattice,
+        uniform_hopping: complex,
+        uniform_onsite_potential: complex,
+        onsite_interaction: complex = None,
+    ) -> "IsingModel":
+        """Set a uniform hopping parameter and on-site potential over the input lattice.
+
+        Args:
+            lattice: Lattice on which the model is defined.
+            uniform_hopping: The hopping parameter. (or uniform_interaction)
+            uniform_onsite_potential: The on-site potential. (or uniform_external_field)
+            onsite_interaction: The strength of the on-site interaction, the Ising model does not
+                have an on-site interaction.
+
+        Returns:
+            The Ising model with uniform parameters.
+
+        Raises:
+            Warning: If the on-site interaction is not None.
+        """
+        if onsite_interaction is not None:
+            raise Warning(
```

Comment:
Since we explicitly ignore the setting down below, we can also just log a warning instead of raising one. If we want to raise one, we should arguably raise a `QiskitNatureError`.
Same applies to `from_parameters` below.

Reply:
For now I've logged all the warnings.